en_US/modules/jwt_authentication.md

```diff
@@ -4,29 +4,36 @@
 
 ## Create module
 
-Open [EMQX Dashboard](http://127.0.0.1:18083/#/modules), click the "Modules" tab on the left, and choose to add:
+Open [EMQX Dashboard](http://127.0.0.1:18083/#/modules), click the "Modules" tab on the left, select "Add Module"
 
-![image-20200928161310952](./assets/modules.png)
+![](./assets/auth_jwt1.png)
 
-Click "Select", and then select the JWT authentication module
+Then select "JWT Authentication" under "Authentication":
 
-![image-20200928141305205](./assets/auth_jwt1.png)
+![](./assets/auth_jwt2.png)
 
-You need to configure the basic connection parameter table such as JWT key, authentication source, and public key (optional).
+JWT authentication provides the following configuration items:
 
-![image-20200928141327836](./assets/auth_jwt2.png)
+1. From. The field that stores the JWT when the client connects, currently supports the selection of username or password.
+2. Secret. The key used when issuing the JWT, which will be used to verify whether the JWT received by EMQX is legal, and is applicable to the JWT issued by the HMAC algorithm.
+3. Pubkey. It will be used to verify whether the JWT received by EMQX is legal, and is applicable to the JWT issued by RSA or ECDSA algorithm.
+4. JWKs Addr. EMQX will periodically query the latest public key list from the JWKS server and use it to verify whether the received JWT is legitimate, and is applicable to JWTs issued by RSA or ECDSA algorithms.
+5. Verify Claims. Whether to verify that the claims in the JWT payload are consistent with the claims.
+6. Claims. A list of claims fields used to verify that the claims in the JWT payload are valid. The most common usage is to add a key-value pair with a key of `username` and a value of `%u`, `%u` as a placeholder that will be replaced at runtime with the Username that the client will actually use when connecting. The replaced value will be used to compare with the value of the same-key claim of the JWT Payload to have a one-to-one correspondence between JWT and Username. The following two placeholders are currently supported in the declaration field list:
+    1. `%u`, will be replaced at runtime with the Username used by the client when connecting
+    2. `%c`, will be replaced at runtime with the Client ID used by the client when connecting
 
-If you need to add custom verification fields, you can refer to the following configuration:
+> Note: If `Secret`, `Pubkey`, and `JWKs Addr` are configured at the same time, EMQX will check the order of secret, public key, and JWKS when verifying JWT.
```

Comment:
Is this true only if all of these are configured or true if any of them are configured? If it happens if more than one of them are configured perhaps a slight change of wording? Perhaps something like:

Note: When verifying JWT the values for `Secret`, `Pubkey`, and `JWKs Addr` are checked in that specific order. Keys with missing values will be ignored.